### PR TITLE
Port the remaining frontend passes to the new pass manager

### PR DESCRIPTION
**Pull request by bsaleil**
This patch ports the remaining frontend passes to the new pass manager.
Some work still needs to be done before we can enable the new pass manager by default for the frontend, so that will be done in a subsequent patch.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit a60d180bf8640e1eb7937a477b479edb1d7fdfcf
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 8e5baae7cfdd885974013337e556e9dc955b6737</li>
<li>SPVGEN: dev 8705e87a8b4aaed55a89f79ed423612ca310bfdd</li>
<li>XGL: dev b39135da435a761ee417609d1106c96fc9da2cc7</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 68e5420f0ba94be9cae6fbac848d746592d03f42</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/221024)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31225/55256 (56.5%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24031/55256 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31225/55256 (56.5%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24031/55256 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31591/55256 (57.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 23665/55256 (42.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34350/55256 (62.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 20906/55256 (37.8%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 92ea5309fad83b8c9c3b07f450a89de199e93244
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 8e5baae7cfdd885974013337e556e9dc955b6737</li>
<li>SPVGEN: dev 8705e87a8b4aaed55a89f79ed423612ca310bfdd</li>
<li>XGL: dev b39135da435a761ee417609d1106c96fc9da2cc7</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 68e5420f0ba94be9cae6fbac848d746592d03f42</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/221024)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31225/55256 (56.5%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24031/55256 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31225/55256 (56.5%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24031/55256 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31591/55256 (57.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 23665/55256 (42.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34350/55256 (62.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 20906/55256 (37.8%)</li>
</ul></details>


</ul></details>

File: llpc/lower/llpcSpirvLower.cpp [hunk 1]
```diff
@@ -178,6 +187,38 @@ void SpirvLower::addPasses(Context *context, ShaderStage stage, lgc::PassManager
 
   // Lower SPIR-V constant folding - must be done before instruction combining pass.
   passMgr.addPass(SpirvLowerMathConstFolding());
+
+  // Lower SPIR-V memory operations
+  passMgr.addPass(SpirvLowerMemoryOp());
+
+  // Remove reduant load/store operations and do minimal optimization
```
Comment (piotrAMD) [inline llpc/lower/llpcSpirvLower.cpp:194]: Typo: reduant.

**Review by piotrAMD — COMMENTED** [review]
LGTM.

**Comment by bsaleil** [comment]
Fixed typo.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 041c5f0f492f1af32ab3456487c0a894e13e38f4
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 8e5baae7cfdd885974013337e556e9dc955b6737</li>
<li>SPVGEN: dev 8705e87a8b4aaed55a89f79ed423612ca310bfdd</li>
<li>XGL: dev b39135da435a761ee417609d1106c96fc9da2cc7</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 68e5420f0ba94be9cae6fbac848d746592d03f42</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/221024)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 31225/55256 (56.5%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24031/55256 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 31225/55256 (56.5%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 24031/55256 (43.5%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 31591/55256 (57.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 23665/55256 (42.8%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34350/55256 (62.2%)</li>
<li>Failed: 0/55256 (0.0%)</li>
<li>Not Supported: 20906/55256 (37.8%)</li>
</ul></details>


</ul></details>

**Review by piotrAMD — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Pull request merged**